### PR TITLE
Add initial Java parser for PathPlanner

### PR DESCRIPTION
**Pull request by bot190**
Defined basic functions and objects to prepare for trajectoryAdds a utility class that can parse a PathPlanner CSV into a Trajectory.

### CODE REVIEW

File: src/main/java/frc/robot/Robot.java [hunk 1]
```diff
@@ -102,12 +102,19 @@ public void robotInit() {
...
         autoChooser.addOption("Slolom", drive.autonomousCommand("Slolom"));
+        autoChooser.addOption("PathPlan Slolom", drive.autonomousCommand("Slolom", true, false));
         autoChooser.addOption("Barrel", drive.autonomousCommand("Barrel"));
+        autoChooser.addOption("PathPlan Barrel", drive.autonomousCommand("Barrel", true, false));
         autoChooser.addOption("CollectBlue", drive.autonomousCommand("CollectBlue"));
         autoChooser.addOption("CollectRed", drive.autonomousCommand("CollectRed"));
         autoChooser.addOption("Bounce", drive.autonomousCommand("Bounce1").andThen(drive.autonomousCommand("Bounce2"))
                 .andThen(drive.autonomousCommand("Bounce3")).andThen(drive.autonomousCommand("Bounce4")));
+        autoChooser.addOption("PathPlan Bounce",
+                drive.autonomousCommand("Bounce1", true, false).andThen(drive.autonomousCommand("Bounce2", true, false))
```
Comment (msoucy) [inline src/main/java/frc/robot/Robot.java]: One thing I noticed is that `x.andThen(y).andThen(z).andThen(q)` can just be done as `x.andThen(y, z, q)` and will produce a more "flat" command structure.

File: src/main/java/frc/robot/subsystems/Drive.java [hunk 2]
```diff
@@ -303,18 +304,27 @@ public void end(final boolean interrupted) {
...
-
-        String trajectoryJSON = "paths/" + trajectoryName + ".wpilib.json";
+    public CommandBase autonomousCommand(String trajectoryName, Boolean resetPose, boolean isJson) {
         Trajectory autoTrajectory = new Trajectory();
-        try {
-            Path trajectoryPath = Filesystem.getDeployDirectory().toPath().resolve(trajectoryJSON);
-            autoTrajectory = TrajectoryUtil.fromPathweaverJson(trajectoryPath);
-        } catch (IOException ex) {
-            DriverStation.reportError("Unable to open trajectory: " + trajectoryJSON, ex.getStackTrace());
+        if (isJson) {
```
Comment (msoucy) [inline src/main/java/frc/robot/subsystems/Drive.java:312]: Do we plan on sticking with just JSON or just CSV, or supporting both long term?

File: src/main/java/frc/robot/utils/PathPlannerUtils.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+            return state;
+        }
+        state.timeSeconds = lineSplit[0];
+        state.velocityMetersPerSecond = lineSplit[1];
+        state.accelerationMetersPerSecondSq = lineSplit[2];
+
+        // Fill in pose information
+        double x = 0, y = 0, rotation = 0;
+        x = lineSplit[3];
+        y = 4.572 - lineSplit[4];
```
Comment (msoucy) [inline src/main/java/frc/robot/utils/PathPlannerUtils.java:30]: Magic number?

File: src/main/java/frc/robot/utils/PathPlannerUtils.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        rotation = -Math.toRadians(lineSplit[5]);
+
+        Pose2d pose = new Pose2d(x, y, new Rotation2d(rotation));
+        state.poseMeters = pose;
+        state.curvatureRadPerMeter = 1 / lineSplit[6];
```
Comment (msoucy) [inline src/main/java/frc/robot/utils/PathPlannerUtils.java:35]: This scared me for a moment. Would you mind doing `1.0 / lineSplit[6]` just so it's more obvious that it's floating-point division?

File: src/main/java/frc/robot/utils/PathPlannerUtils.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+        state.accelerationMetersPerSecondSq = lineSplit[2];
+
+        // Fill in pose information
+        double x = 0, y = 0, rotation = 0;
+        x = lineSplit[3];
+        y = 4.572 - lineSplit[4];
+        rotation = -Math.toRadians(lineSplit[5]);
+
+        Pose2d pose = new Pose2d(x, y, new Rotation2d(rotation));
+        state.poseMeters = pose;
```
Comment (msoucy) [inline src/main/java/frc/robot/utils/PathPlannerUtils.java:34]: Why does this use a temporary variable?

File: src/main/java/frc/robot/utils/PathPlannerUtils.java [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+        state.curvatureRadPerMeter = 1 / lineSplit[6];
+        return state;
+    }
+
+    public static Trajectory fromPathPlannerCSV(Path path) throws IOException {
```
Comment (msoucy) [inline src/main/java/frc/robot/utils/PathPlannerUtils.java:39]: Can has documentation? (Here and elsewhere)

Basically if it's going to go in chopshoplib someday it should have documentation early

**Review by msoucy — CHANGES_REQUESTED** [review]

**Comment by msoucy** [comment]
Status? Rebase?

**Comment by bot190** [comment]
The only part of this we might keep is `PathPlannerUtils.java` which should go to chopshoplib. Closing this.

**Pull request closed without merging**